internal/gomod/dependencies.go

```diff
@@ -95,15 +107,76 @@ func parseGoListOutput(output, rootVersion string) (map[string]Module, error) {
 			module.Version = rootVersion
 		}
 
-		dependencies[importPath] = Module{
+		dependencies[importPath] = GoModule{
 			Name:    module.Name,
 			Version: cleanVersion(module.Version),
 		}
 	}
 
+	if goVersion == "" {
+		return nil, errors.New("must have a valid go version in go mod file.")
+	}
+	setGolangDependency(dependencies, goVersion)
+
 	return dependencies, nil
 }
 
+// The repository to find the source code for golang.
+var golangRepository = "github.com/golang/go"
+
+func setGolangDependency(dependencies map[string]GoModule, goVersion string) {
+	dependencies[golangRepository] = GoModule{
+		Name: golangRepository,
+
+		// The reason we prefix version with "go" is because in golang/go, all the release
+		// tags are prefixed with "go".
+		//
+		// The corresponding tag of Golang 1.16 is "go1.16"
+		Version: "go" + goVersion,
+	}
+
+}
+
+func GetGolangDependency(dependencies map[string]GoModule) GoModule {
+	return dependencies[golangRepository]
+}
+
+// IsStandardlibPackge checks whether a particular package is a standard
+// library package.
+func IsStandardlibPackge(pkg string) bool {
+	// No standard library packages have a domain name associated with them.
+	//
+	// So if we see any "." in the package name, then we know that they
+	// represent some non-standard library package.
+	if strings.Contains(pkg, ".") {
+		return false
+	}
+
+	// No standard library packages are prefixed with "_".
+	// This can be used to denote testing packages.
+	if strings.HasPrefix(pkg, "_") {
+		return false
+	}
+
+	return true
```

Comment:
Just to really ruin this for ya, its entirely possible and legit for non-stdlib fully-qualified import path to _not_ be a domain name. Youd usually only see this in application repos, as you cant `go get` them for use as libraries

Reply:
Conferring with a Go compiler dev, a possibility would be to invoke `go list std` and seeing if the import name is in that list or not

Reply:
oh, `go list std` is way easier. Only thing would be multiple different versions, but we can update that manually over time or something.